core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -558,7 +555,16 @@ private ClientFactoryOptions buildOptions() {
             return ClientFactoryOption.ADDRESS_RESOLVER_GROUP_FACTORY.newValue(addressResolverGroupFactory);
         });
 
-        return ClientFactoryOptions.of(options.values());
+        final ClientFactoryOptions newOptions = ClientFactoryOptions.of(options.values());
+        final long idleTimeoutMillis = newOptions.idleTimeoutMillis();
+        final long pingIntervalMillis = newOptions.pingIntervalMillis();
+        if (idleTimeoutMillis > 0 && pingIntervalMillis > 0) {
+            checkArgument(idleTimeoutMillis > pingIntervalMillis,
+                          "idleTimeoutMillis: %s, pingIntervalMillis: %s " +
+                          "(expected: idleTimeoutMillis > pingIntervalMillis)",
+                          idleTimeoutMillis, pingIntervalMillis);
+        }
+        return newOptions;
```

Comment:
A user might find this inconvenient because he or she has to update the both settings when he or she wants to disable the ping and reduce the idle timeout. How about automatically disabling ping interval when it's greater than idle timeout?

Reply:
SGTM, disabled ping interval automatically and updated Javadoc.